### PR TITLE
Nept 293: [DIGIT Service Catalogue] Issue with ecas group sync after upgrade to 2.2

### PR DESCRIPTION
**Pull request by gillesdeudon**

### CODE REVIEW

File: profiles/common/modules/custom/ecas/ecas_group_sync/ecas_group_sync.module [hunk 1]
```diff
@@ -50,50 +50,106 @@ function ecas_group_sync_menu() {
...
+ *   The user's DG identifier; e.g. DIGIT.
+ * @param string $sync_type
+ *   The type of roles implied in the mappung; 'role' for DG/Drupal roles
+ *   mapping, 'group' for for DG/OG roles.
+ *
+ * @return array|bool
+ *    An array of objects containing rules data; otherwise FALSE.
+ */
+function ecas_group_sync_get_ecas_sync_rules($dg_id, $sync_type = 'role') {
+  $sync_rules = &drupal_static(__FUNCTION__);
```
Comment (gillesdeudon) [inline profiles/common/modules/custom/ecas/ecas_group_sync/ecas_group_sync.module]: I know but when I designed this function, I thought about the future.
If I have the opportunity and the time, I prefer to foresee the future; it is already that that we will not need to do later.

File: profiles/common/modules/custom/ecas/ecas_group_sync/ecas_group_sync.admin.inc [hunk 2]
```diff
@@ -149,6 +149,9 @@ function ecas_group_sync_mapping_form_submit($form, &$form_state) {
     }
   }
 
+  // Clearing the cache.
+  cache_clear_all('ecas_group_sync_rules', 'cache');
```
Comment (gillesdeudon) [inline profiles/common/modules/custom/ecas/ecas_group_sync/ecas_group_sync.admin.inc]: Ok.

File: profiles/common/modules/custom/ecas/includes/ecas.inc [hunk 3]
```diff
@@ -732,3 +732,37 @@ function ecas_create_user($ecas_name, $defaults = array()) {
...
+    $department = array('full_name' => $user_info['cas:departmentNumber']);
+    $organizations = explode('.', $user_info['cas:departmentNumber']);
+
+    $department += array(
+      'dg' => $organizations[0],
+      'directorate' => (isset($organizations[1])) ?: FALSE,
+      'unit' => (isset($organizations[2])) ?: FALSE,
+      'sub_unit' => (isset($organizations[3])) ?: FALSE,
+    );
+
```
Comment (gillesdeudon) [inline profiles/common/modules/custom/ecas/includes/ecas.inc]: OK.

**Comment by dxvargas** [comment]
👍

**Comment by imanoleguskiza** [comment]
👍

**Pull request merged**